Sources/GATTBodyCompositionMeasurement.swift

```diff
@@ -0,0 +1,720 @@
+        case basalMetabolism = 0b1000
+        
+        /// Muscle Percentage present
+        case musclePercentage = 0b10000
+        
+        /// Muscle Mass present
+        case muscleMass = 0b100000
+        
+        /// Fat Free Mass present
+        case fatFreeMass = 0b1000000
+        
+        /// Soft Lean Mass present
+        case softLeanMass = 0b10000000
+        
+        /// Body Water Mass present
+        case bodyWaterMass = 0b100000000
+        
+        /// Impedance present
+        case impedance = 0b1000000000
+        
+        /// Weight present
+        case weight = 0b10000000000
+        
+        /// Height present
+        case height = 0b100000000000
+        
+        /// Multiple Packet Measurement
+        case multiplePacket = 0b1000000000000
+        
+        public static var all: Set<Flag> = [
+             .measurementUnitSI,
+             .measurementUnitImperial,
+             .timestamp,
+             .userID,
+             .basalMetabolism,
+             .musclePercentage,
+             .muscleMass,
+             .fatFreeMass,
+             .softLeanMass,
+             .bodyWaterMass,
+             .impedance,
+             .weight,
+             .height,
+             .multiplePacket
+        ]
+    }
+    
+    public enum MassUnit: UInt16 {
+        
+        /// Mass kilogram
+        case kilogram = 0x2702
+        
+        /// Mass pound
+        case pound = 0x27B8
+        
+        public init?(unit: UnitIdentifier) {
+            
+            self.init(rawValue: unit.rawValue)
+        }
+        
+        public var unit: UnitIdentifier {
+            
+            return UnitIdentifier(rawValue: rawValue)
+        }
+    }
+    
+    public enum LengthUnit: UInt16 {
+        
+        /// Length metre
+        case metre = 0x2701
+        
+        /// length inch
+        case inch = 0x27A2
+        
+        public init?(unit: UnitIdentifier) {
+            
+            self.init(rawValue: unit.rawValue)
+        }
+        
+        public var unit: UnitIdentifier {
+            
+            return UnitIdentifier(rawValue: rawValue)
+        }
+    }
+    
+    public struct BodyMass {
+        
+        public typealias MassUnit = GATTBodyCompositionMeasurement.MassUnit
+        
+        internal static let length = MemoryLayout<UInt16>.size
+        
+        public var unit: MassUnit
+        
+        public var rawValue: UInt16
+        
+        public init?(rawValue value: UInt16, unit: MassUnit) {
+            
+            self.rawValue = value
+            self.unit = unit
+        }
+    }
+    
+    public struct BodyLength {
+        
+        public typealias LengthUnit = GATTBodyCompositionMeasurement.LengthUnit
+        
+        internal static let length = MemoryLayout<UInt16>.size
+        
+        public var unit: LengthUnit
+        
+        public var rawValue: UInt16
+        
+        public init?(rawValue value: UInt16, unit: LengthUnit) {
```

Comment:
make non-fallible initializer public init(rawValue: UInt16, unit: MassUnit)

Reply:
@JurjDev @carlos21 I think we should rename the `rawValue` to just `value` because we aren't implementing the `RawRepresentable` protocol which is really for a struct with a single property, not multiple.